pulp_ansible/app/sigstore/tasks/sigstore_signature.py

```diff
@@ -0,0 +1,295 @@
+"""
+Tasks related to Sigstore content signature and verification.
+"""
+
+import aiofiles
+import asyncio
+import logging
+import os
+import re
+import tarfile
+import tempfile
+import cryptography.x509 as x509
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import ec
+from cryptography.x509.oid import NameOID
+from distlib.manifest import DistlibException
+from gettext import gettext as _
+
+from ansible_sign.checksum import (
+    ChecksumFile,
+)
+from ansible_sign.checksum.differ import DistlibManifestChecksumFileExistenceDiffer
+
+from sigstore.verify.models import VerificationFailure
+from sigstore._internal.oidc import Identity
+from sigstore._internal.sct import verify_sct
+from sigstore._utils import sha256_streaming
+from sigstore_protobuf_specs.dev.sigstore.bundle.v1 import Bundle
+
+from pulpcore.plugin.stages import (
+    DeclarativeContent,
+    Stage,
+)
+
+from pulp_ansible.app.tasks.signature import SigningDeclarativeVersion
+
+from django.conf import settings
+from django.core.files.storage import default_storage as storage
+from pulpcore.plugin.models import ProgressReport
+from pulpcore.plugin.sync import sync_to_async_iterable, sync_to_async
+from pulp_ansible.app.sigstore.exceptions import VerificationFailureException
+
+from pulp_ansible.app.models import (
+    CollectionVersion,
+    CollectionVersionSigstoreSignature,
+    SigstoreSigningService,
+)
+
+log = logging.getLogger(__name__)
+
+
+def _generate_checksum_manifest(collection_version):
+    """
+    Generate a file containing the sha256 hashes of all the collection elements.
+    """
+    cartifact = collection_version.contentartifact_set.select_related("artifact").first()
+    artifact_name = cartifact.artifact.file.name
+    artifact_file = storage.open(artifact_name)
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        differ = DistlibManifestChecksumFileExistenceDiffer
+        checksum = ChecksumFile(tempdir, differ=differ)
+        with tarfile.open(fileobj=artifact_file, mode="r") as tar:
+            tar.extractall(path=tempdir)
+
+            try:
+                manifest = checksum.generate_gnu_style()
```

Comment:
This seems potentially really slow. We already have the checksum file `MANIFEST.IN` in the tarball, can we not just extract it and pass it along instead of re-calculating the whole thing?

Reply:
That could be done, but initially I went with this approach to conserve compatibility with [`ansible-sign`](https://github.com/ansible/ansible-sign/blob/9e28a8ad65d89fceebc5d6a2a1ebb3e37f142512/src/ansible_sign/cli.py#L161) as it will also be used to sign Ansible artifacts with Sigstore in the future.

Reply:
What is the status of `ansible-sign`? I see it below as a new dependency. Do we still need to have this code or can we call `ansible-sign`'s code instead?